lib/ingester.js

```diff
@@ -59,12 +60,15 @@ class Ingester {
 
     async addAsset (asset) {
         const assets = asset.flattenAssetTree();
-        this.assetsCount.total += assets.length;
+
+        for (const a of assets) {
+            await a.finish();
+        }
+
         if (!asset.isValidTree()) {
-            this.assetsCount.failed += assets.length;
             logger.error(
-                `Asset ${asset.canonicalURI} has failed to ingest or it has failed children`);
-            return;
+                `Asset ${asset.metadata.canonicalURI}` +
+                'has failed to ingest or it has failed children');
```

Comment:
Maybe for this case, we should mark all the assets from the tree as invalid. Because otherwise we're gonna ingest articles with failed relatives.

Reply:
Yes, true. This is the biggest issue in the PR. Thanks for reviewing.